x-pack/plugins/embeddable_enhanced/public/drilldowns/url_drilldown/url_drilldown_scope.ts

```diff
@@ -154,18 +137,23 @@ async function getEventScopeFromRangeSelectTriggerContext(
   });
 }
 
-async function getEventScopeFromValueClickTriggerContext(
-  eventScopeInput: ValueClickContext,
-  deps: { getOpenModal: () => Promise<OverlayStart['openModal']> },
-  opts: { allowPrompts: boolean } = { allowPrompts: false }
-): Promise<ValueClickTriggerEventScope> {
+function getEventScopeFromValueClickTriggerContext(
+  eventScopeInput: ValueClickContext
+): ValueClickTriggerEventScope {
   const negate = eventScopeInput.data.negate ?? false;
-  const point = await getSingleValue(eventScopeInput.data.data, deps, opts);
-  const { key, value } = getKeyValueFromPoint(point);
+  const points = eventScopeInput.data.data.map(({ table, value, column: columnIndex }) => {
+    const column = table.columns[columnIndex];
+    return {
+      value: toPrimitiveOrUndefined(value) as string | number | boolean,
+      key: toPrimitiveOrUndefined(column?.meta?.aggConfigParams?.field) as string | undefined,
```

Comment:
Question: what do you think making it `toPrimitiveOrNull` instead of `toPrimitiveOrUndefined`. The reason being that `null` is a valid JSON value, for example, if somebody will do `{{json event}}`, the undefined values will disappear.

![image](https://user-images.githubusercontent.com/9773803/93353010-ba440400-f83b-11ea-904a-c8830c1b03e7.png)

Reply:
Thanks for pointing this out, I will review `null` case and open a separate pr. I think you are right that `null` should stay as `null`.

The idea behind `undefined` values that those are removed from the result object afterwards. This is needed for `handlebars` validations to throw errors when when missing values is accesses.

Reply:
Improved in this PR 👍